hashin.py

```diff
@@ -228,6 +219,69 @@ def run_packages(
         if verbose:
             _verbose("Editing", file)
 
+    return 0
+
+
+class InteractiveAll(Exception):
+    """When the user wants to say yes to ALL package updates."""
+
+
+class InteractiveQuit(Exception):
+    """When the user wants to stop the interactive update questions entirely."""
+
+
+def interactive_upgrade_request(package, old_version, new_version, print_header=False):
+    def print_version(v):
+        return str(v).replace("==", "").ljust(15)
+
+    if print_header:
+        print(
+            "PACKAGE".ljust(30),
+            print_version("YOUR VERSION"),
+            print_version("NEW VERSION"),
+        )
+
+    def print_line(checkbox=None):
+        if checkbox is None:
+            checkboxed = "?"
+        elif checkbox:
+            checkboxed = "✓"
+        else:
+            checkboxed = "✘"
+        print(
+            package.ljust(30),
+            print_version(old_version),
+            print_version(new_version),
+            checkboxed,
+        )
+
+    print_line()
+
+    def clear_line():
+        sys.stdout.write("\033[F")  # Cursor up one line
+        sys.stdout.write("\033[K")  # Clear to the end of line
+
+    def ask():
+        answer = input("Update? [Y/n/a/q]: ").lower().strip()
```

Comment:
As a nifty improvement, it would be cool to add "?" to the list of options, and if ? or an invalid choice is given, it prints a longer version of this prompt out, like:

```
y - Include this upgrade (default)
n - Skip this upgrade
a - Include this and all following upgrades
q - Skip this and all following upgrades
? - Print this help
```

You can see a similar UI in `git add -i`, which prompts you to include, skip, edit, etc chunks of files.

Reply:
+1